<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>4000</width>
    <height>2000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #2c3e50; 
    color: #ecf0f1; 
    font-size: 10px;
}

QPushButton {
    background-color: #1abc9c; 
    color: white;
    font-size: 16px;
    font-weight: bold;
    border: none;
    border-radius: 8px;
    padding: 10px 20px;
}

QPushButton:hover {
    background-color: #16a085; 
}

QPushButton:pressed {
    background-color: #149174;
}

QPushButton:disabled {
    background-color: #95a5a6; 
    color: #7f8c8d;
}

QFrame#screen {
    background-color: #34495e; 
    border: 3px solid #1abc9c; 
    border-radius: 5px;
}

QComboBox {
    background-color: #34495e; 
    color: #ecf0f1; 
    border: 2px solid #1abc9c; 
    border-radius: 5px;
	padding: 5px;
}

QComboBox::drop-down {
    background: #1abc9c;
    border: none;
}

QComboBox QAbstractItemView {
    background-color: #2c3e50;
    color: #ecf0f1;
	border: 1px solid #1abc9c;
}

QSpinBox, QDoubleSpinBox {
    background-color: #34495e; 
    color: #ecf0f1;
   
}

QLabel {
    color: #ecf0f1;
}

QMenuBar {
    background-color: #1abc9c;
    color: white;
    font-weight: bold;
}

QMenuBar::item:selected {
    background-color: #16a085;
}

QStatusBar {
    background-color: #34495e;
    color: #ecf0f1;
    font-size: 12px;
}

QProgressBar {
    background-color: #34495e;  
    border: 2px solid #1abc9c;  
    border-radius: 8px;         
    text-align: center;         
    color: #ecf0f1;             
    font-weight: bold;          
	font-size: 25px;
}

QProgressBar::chunk {
    background-color: #1abc9c;  
    border-radius: 8px;         
}

QLabel#loadingScreen {
    background-color: #34495e;  
    color: #ecf0f1;            
    border: 2px solid #1abc9c; 
    border-radius: 8px;        
    padding: 10px;              
    font-size: 14px;           
    font-weight: bold;          
    text-align: center;     
}


</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="screen">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>1270</width>
      <height>940</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>OpenHandCursor</cursorShape>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 0);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>5</number>
    </property>
    <property name="midLineWidth">
     <number>2</number>
    </property>
    <widget class="QProgressBar" name="progressBar">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>350</y>
       <width>118</width>
       <height>23</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="textVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="loadingScreen">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>250</y>
       <width>151</width>
       <height>71</height>
      </rect>
     </property>
     <property name="text">
      <string>TRWA RYSOWANIE OBRAZU...</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>1320</x>
      <y>40</y>
      <width>492</width>
      <height>871</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="rightSideLayout">
     <item>
      <layout class="QVBoxLayout" name="panelControlLayout">
       <property name="spacing">
        <number>20</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="threadsLayout">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Ilość wątków</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox">
           <property name="minimum">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="settingsLayout">
         <item>
          <layout class="QHBoxLayout" name="iterationLayout">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Liczba iteracji</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_2">
             <property name="minimum">
              <number>50</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="resolutionLayout">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Rozdzielczość obrazu (w px)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="widthLayout">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Szerokość</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_4">
               <property name="minimum">
                <number>1270</number>
               </property>
               <property name="maximum">
                <number>3000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="heightLayout">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Wysokość</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_3">
               <property name="minimum">
                <number>940</number>
               </property>
               <property name="maximum">
                <number>3000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="bordersLayout">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Granice wycinka płaszczyzny zespolonej</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QGridLayout" name="borderLayout">
           <item row="2" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>y_min</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
             <property name="minimum">
              <double>-5.000000000000000</double>
             </property>
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>-1.500000000000000</double>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>x_min</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
             <property name="minimum">
              <double>-5.000000000000000</double>
             </property>
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.500000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>x_max</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
             <property name="minimum">
              <double>-5.000000000000000</double>
             </property>
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>-1.500000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>y_max</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
             <property name="minimum">
              <double>-5.000000000000000</double>
             </property>
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.500000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="imaginaryNumberLayout">
         <item>
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Liczba zespolona</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <layout class="QHBoxLayout" name="imagLayout">
             <item>
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Część rzeczywista</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
               <property name="minimum">
                <double>-2.000000000000000</double>
               </property>
               <property name="maximum">
                <double>2.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.050000000000000</double>
               </property>
               <property name="value">
                <double>-0.700000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="realLayout">
             <item>
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Część urojona</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox">
               <property name="minimum">
                <double>-2.000000000000000</double>
               </property>
               <property name="maximum">
                <double>2.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.050000000000000</double>
               </property>
               <property name="value">
                <double>0.270000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="bottomLayout">
         <item>
          <layout class="QHBoxLayout" name="comboBoxLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Wybierz paletę:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox">
             <item>
              <property name="text">
               <string>Skala szarości</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>HSV</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Gładki gradient kolorów</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów 2</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów 3</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów 4</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów 5</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Interpolowana mapa kolorów 6</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="buttonsLayout">
           <item row="0" column="0">
            <widget class="QPushButton" name="generateImagePushButton">
             <property name="font">
              <font>
               <pointsize>-1</pointsize>
               <bold>true</bold>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string>Generuj obraz</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="clearImagePushButton">
             <property name="text">
              <string>Czyść obraz</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="resetValuesPushButton">
             <property name="font">
              <font>
               <pointsize>-1</pointsize>
               <bold>true</bold>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string>Resetuj wartości spinboxów</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="strongScalingTestPushButton">
             <property name="font">
              <font>
               <pointsize>-1</pointsize>
               <bold>true</bold>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string>Test silnej skalowalności</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="consoleLayout">
       <item>
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Dane z konsoli</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="plainTextEdit">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="font">
          <font>
           <family>Courier</family>
           <pointsize>12</pointsize>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>4000</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
